lib/drivers/ism43340/ism43340.cpp

```diff
@@ -0,0 +1,270 @@
+    }
+}
+
+int Ism43340::writetospi(std::string* command) {
+    while (dataready.read() != 1) {
+    }
+
+    uint8_t lastc = 0;
+    radiospi.chipselect();
+    for (int i = 0; i < command->size(); i++) {
+        uint8_t c = command[i];
+        if (lastc != 0) {
+            uint16_t packet = (c << 8) | lastc;
+            radiospi.m_spi->write(packet);
+            lastc = 0;
+        } else {
+            lastc = c;
+        }
+    }
+    radiospi.chipdeselect();
+    return 0;
+}
+
+int Ism43340::readfromspi(uint8* readbuffer, uint32* readlength) {
+    while (dataready.read() != 1) {
+    }
+
+    packetcount = 0;
+    radiospi.chipselect();
+    while (dataready.read() != 0) {
+        uint16_t data = radiospi.m_spi->write(0x0a0a);
+        readbuffer[packetcount] = (uint8_t)(data);
+        readbuffer[packetcount + 1] = (uint8_t)(data >> 8);
+        packetcount += 2;
+    }
+    radiospi.chipdeselect();
+    return 0;
+}
+
+
+/* COMMAND SET DEFINITIONS */
+
+//Check prompt
+inline checkPrompt () {
+    // read the initial prompt into a buffer
+    readfromspi();
+    // check the buffer for the response sequence of the prompt (Not the same as the OK)
+    // you could probably have read spi return an int as a status code for both cant decide
+}
+
+// This command will probably not be used but is here as an example command
+// Set Human Readable
+inline setHumanReadable() {
+    std::string compositeString(CMD_SET_HUMAN_READABLE);
+
+    // formats command adding /r and /n when needed
+    formatCommand(&compositeString);
+
+    // write composite string to the radio
+    writetospi(&compositeString);
+    // read to global buffer
+    readfromspi();
+}
+
+//Set Machine Readable
+
+//Connect to access point
+
+//Set SSID
+
+//Set network security type
+
+//Set DHCP usage
+
+//Disconnect from network
+
+//Show connection status
+
+// This one is here as an example as well
+//Show network settings
+inline showNetworkSettings() {
+    std::string compositeString(CMD_SHOW_NETWORK_SETTINGS);
+
+    // formats command adding /r and /n when needed
+    formatCommand(&compositeString);
+
+    // write composite string to the radio
+    writetospi(&compositeString);
+    // read to global buffer. This needs to be returned to the user
+    readfromspi();
+    // the returned buffer from this is of unknown length. However it is well bounded by the length of possible network information
+}
+
+
+int32_t Ism43340::sendPacket(const rtp::Packet* pkt) {
+    // Return failutre if not initialized
+    if (!m_isInit) return COMM_FAILURE;
+
+
+    // dont know how much of this needs to be kept. Soccer may actually want the split between body and head of the sent packet. I dont know
+    BufferT txBuffer;
+
+    const auto bufferSize = 9 + pkt->size() + 2;
+    txBuffer.reserve(bufferSize);
+
+    // MAC layer header for Decawave
+    txBuffer.insert(txBuffer.end(),
+                    {0x41, 0x88, 0x00, 0xCA, 0xDE, pkt->header.address, 0x00,
```

Comment:
I like defines for these too

Reply:
This is a DWM thing, I think it'll go away. Note the comment says MAC layer for DWM